docs/source/development/start/setup.rst

```diff
@@ -0,0 +1,165 @@
+
+.. code:: bash
+
+    git remote add upstream https://github.com/LineaLabs/lineapy.git
+
+Setting up Virtual Environment
+******************************
+
+When done cloning, move into the repo and initiate a virtual environment:
+
+.. code:: bash
+
+    cd <FORKED-REPO-NAME>
+    python -m venv env
+
+where ``<FORKED-REPO-NAME>`` is ``lineapy`` unless modified during/after forking.
+
+This creates ``env/`` subfolder inside the repo, and you can activate the associated virtual environment like so:
+
+.. code:: bash
+
+    source env/bin/activate
+
+To deactivate the virtual environment, you can type:
+
+.. code:: bash
+
+    deactivate
+
+Installing Dependencies
+***********************
+
+Once the virtual environment is activated, install packages necessary for running and developing ``lineapy``:
+
+.. code:: bash
+
+    pip install --upgrade pip
+    pip install --upgrade setuptools
+    pip install -r requirements.txt
+    pip install -e '.[dev]'
+
+Note that this may take a while to complete.
+
+Running LineaPy
+---------------
+
+During development, you may want to check how your changes affect the behavior of the package.
+LineaPy supports several interfaces to run on, and the relevant instructions are documented :ref:`here <interfaces>`.
+Please select the most convenient option for you.
+
+.. note::
+
+    When running LineaPy, make sure that the virtual environment has been activated.
+
+Development
+-----------
+
+Making Changes
+**************
+
+With development dependencies installed, you are now ready to contribute to the source code!
+First, make a separate branch for your development work. Please use an informative name so that
+others can get good sense of what your changes are about.
+
+.. code:: bash
+
+    git checkout -b <NEW-BRANCH-NAME>
+
+After making changes you desire, save them to your development branch:
+
+.. code:: bash
+
+    git add <PATH-TO-CHANGED-FILE>
+    git commit -m "<COMMIT-MESSAGE>"
+
+.. note::
+
+    To learn more about saving changes in Git, check this `tutorial <https://www.atlassian.com/git/tutorials/saving-changes>`_.
+
+Note that these changes have been saved only locally at this point, and you need to "push" them to your forked repo on GitHub:
+
+.. code:: bash
+
+    git push
+
+If the new (development) branch has not been pushed before, you will need to create its counterpart on GitHub with:
+
+.. code:: bash
+
+    git push --set-upstream origin <NEW-BRANCH-NAME>
+
+Testing Changes
+***************
+
+Testing is an important part of ``lineapy``'s development as it ensures that all features stay functional after changes.
+Hence, we strongly recommend you add tests for changes you introduce (check this :ref:`tutorial <add_test>` for adding tests).
+
+To run all tests (beware that this may take a while to complete):
+
+.. code:: bash
+
+    pytest tests
+
+Or, to run a particular test (e.g., one that you added/modified):
+
+.. code:: bash
+
+    pytest <PATH-TO-TEST-FILE>
```

Comment:
I like adding more examples here such as 

Running all tests in a folder
`pytest tests/unit/plugins` 

Running a particular test file
`pytest tests/unit/plugins/test_writer.py` 

Running a particular test
`pytest tests/unit/plugins/test_writer.py::test_pipeline_generation` 

Running a single test with one parametrization
`pytest tests/unit/plugins/test_writer.py::test_pipeline_generation[script_pipeline_a0_b0]`

Reply:
Good suggestion, acting on it.